### PR TITLE
SQ-82/Card views

### PR DESCRIPTION
**Pull request by rock3r**
Cards! With shadows! And magic ✨!

* Create proper cards for schedule
* MAGIC to create shadows matching specs (not straight-up elevation, but some trickery)
   * Do not try this at home
* More abusing of Outline
   * Seriously, don't do this without adult supervision
* Tweak stuff while pairing with @DigitalPencils 
* Get really frustrated with Sketch because its SVG exporting capabilities suck, then Affinity Designer saves the day
* Stuff? Stuff.

Still TODO:
 * Wire up a bunch of the data which we don't have yet
 * Support multiple speakers (design still missing)
 * Fix tabstrip
    * Font and capitalisation
    * Selection indicator colour
    * Date formatting (should be full day)

image

### CODE REVIEW

File: app/src/main/java/net/squanchy/support/widget/CardLayout.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    private void ensureHasNecessaryAttributes(TypedArray a) {
+        if (!a.hasValue(R.styleable.CardLayout_cardInsetHorizontal)) {
+            throw new IllegalStateException("Missing the cardInsetHorizontal attribute in the style");
+        }
+        if (!a.hasValue(R.styleable.CardLayout_cardInsetTop)) {
+            throw new IllegalStateException("Missing the cardInsetTop attribute in the style");
+        }
+        if (!a.hasValue(R.styleable.CardLayout_cardCornerRadius)) {
+            throw new IllegalStateException("Missing the cardCornerRadius attribute in the style");
+        }
```
Comment (danybony) [inline app/src/main/java/net/squanchy/support/widget/CardLayout.java]: Did you forget about `insetBottom` or is it not necessary?

**Review by danybony — COMMENTED** [review]
Just one question and then good to go!

File: app/src/main/java/net/squanchy/support/widget/CardLayout.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/support/widget/CardLayout.java]: Forgot it :D was a last minute addition

File: app/src/main/res/values/attrs.xml [hunk 2]
```diff
@@ -7,4 +7,23 @@
...
+  <declare-styleable name="CardLayout">
+    <attr name="cardCornerRadius" format="dimension" />
+    <attr name="cardInsetHorizontal" format="dimension" />
+    <attr name="cardInsetTop" format="dimension" />
+    <attr name="cardInsetBottom" format="dimension" />
+  </declare-styleable>
+
+  <attr name="cardViewDefaultStyle" format="reference" />
+
+  <style name="Widget.Squanchy.CardLayout" parent="None">
```
Comment (danybony) [inline app/src/main/res/values/attrs.xml:19]: Mmm this should be in `styles.xml`

Comment (rock3r) [reply app/src/main/res/values/attrs.xml:19]: That's a default style (`Widget.`), didn't want to mix it with "real" ones. If we end up having more base styles I'll put them in a separate file

File: app/src/main/java/net/squanchy/support/widget/CardLayout.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/support/widget/CardLayout.java]: 7ba6f6e8fc3dce46a5d49761acccbffbf7e9e0af

**Review by danybony — APPROVED** [review]

**Pull request merged**